packages/asset-swapper/src/utils/market_operation_utils/path.ts

```diff
@@ -92,10 +100,14 @@ export class Path {
         // By prepending native paths to the front they cannot split on-chain sources and incur
         // an additional protocol fee. I.e [Uniswap,Native,Kyber] becomes [Native,Uniswap,Kyber]
         // In the previous step we dropped any hanging Native partial fills, as to not fully fill
-        const nativeFills = this.fills.filter(f => f.source === ERC20BridgeSource.Native);
-        this.fills = [...nativeFills.filter(f => f !== lastNativeFillIfExists), ...fallback.fills];
-        // Recompute the source flags
-        this.sourceFlags = this.fills.reduce((flags, fill) => flags | fill.flags, BigInt(0));
+        const nativeFills = this.fills.filter(f => f.source === ERC20BridgeSource.Native && f !== lastNativeFillIfExists);
+        this.fills = [];
+        this.sourceFlags = BigInt(0);
+        this._fillsById = {};
+        this._numDistinctFills = 0;
```

Comment:
Just to double-check, it's intentional to reset this to 0 and not increment it when adding fills below?

Reply:
well it was supposed to be updated in `_addFill()` but looks like that particular line of code never made it out of my head :facepalm: